util/lua/lua.go

```diff
@@ -426,11 +426,14 @@ func GetConfigMapKey(gvk schema.GroupVersionKind) string {
 	return fmt.Sprintf("%s/%s", gvk.Group, gvk.Kind)
 }
 
+// GetWildcardConfigMapKey returns the first encountered resource override which matches the wildcard and has a
+// non-empty health script. Having multiple wildcards with non-empty health checks that can match the GVK is
+// non-deterministic.
 func GetWildcardConfigMapKey(vm VM, gvk schema.GroupVersionKind) string {
```

Comment:
I know it's a bit scope-creepy, but for simplicity I'd:

1) Change the function signature to `(overrides map[string]appv1.ResourceOverride, gvk schema.GroupVersionKind)`. It's weird to me that we accept the whole `vm` but only use one field. Makes it harder to tell from the caller's perspective what the function does.
2) Return the health check instead of the map key. It's weird that the function does the work to find the key and then the caller immediately has to double-check that the key is present. We already found the health script, we should just return it.
3) Rename the function to reflect the above two changes.
4) Make the function private. There's no need for it to be public, it's not really library code, and it just gets us into the situation of fretting about public API changes.

Personally I'm not worried about public API changes. We don't advertise any of this as a library, and it's vanishingly unlikely that anyone's using it that way.

Reply:
Done - this also needs to return `useOpenLibs` (not that it is used so much in practice presumably?), so I return that along with the healthcheck itself.